test-framework/junit5/src/main/java/io/quarkus/test/junit/nativeimage/BuildOutputExtension.java

```diff
@@ -0,0 +1,138 @@
+package io.quarkus.test.junit.nativeimage;
+
+import static io.quarkus.test.junit.IntegrationTestUtil.readQuarkusArtifactProperties;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Locale;
+import java.util.Properties;
+
+import jakarta.json.Json;
+import jakarta.json.JsonObject;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Assumptions;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import io.quarkus.deployment.pkg.steps.GraalVM;
+
+/**
+ * This is a general utility to assert via
+ * unit testing how many classes, methods, objects etc. have been included in a native-image.
+ * <p>
+ * For detailed information and explanations on the build output, visit
+ * <a href="https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md">the upstream GraalVM
+ * documentation</a>.
+ */
+public class BuildOutputExtension implements BeforeAllCallback {
```

Comment:
Minor nitpick - can we improve the name of this? Perhaps `NativeBuildOutputExtension`?

Reply:
Done